rllib/utils/actor_manager.py

```diff
@@ -0,0 +1,324 @@
+            fetch_local=True,
+        )
+
+        if timeout and len(ready) < len(remote_calls):
+            logger.error(
+                f"Failed to wait for {len(remote_calls) - len(ready)} remote calls."
+            )
+
+        # Remote data should already be fetched to local object store at this point.
+        remote_results = []
+        for i, r in enumerate(ready):
+            actor_idx = remote_actor_indices[i]
+            try:
+                remote_results.append(ray.get(r))
+                # Able to fetch return value. Mark this actor active if necessary.
+                if not self.is_actor_healthy(actor_idx):
+                    print(f"brining actor {actor_idx} back into service.")
+                self.set_actor_state(actor_idx, healthy=True)
+            except RayActorError as e:
+                remote_results.append(e)
+                # Take this actor out of service and wait for Ray Core to recover it.
+                if self.is_actor_healthy(actor_idx):
+                    print(
+                        f"Ray error, taking actor {actor_idx} out of service. {str(e)}"
+                    )
+                self.set_actor_state(actor_idx, healthy=False)
+            except Exception as e:
+                # Application error. Return Exception as-is to the caller.
+                remote_results.append(e)
+
+        return ready, remote_results
+
+    @DeveloperAPI
+    def foreach_actor(
+        self,
+        func: Callable[[T], T],
+        *,
+        healthy_only=True,
+        remote_actor_indices: List[int] = None,
+        timeout_seconds=None,
+    ) -> List[T]:
+        """Calls the given function with each actor instance as arg.
+
+        Args:
+            func: The function to call for each actor (as only arg).
+            healthy_only: Apply func on known active actors only.
+            remote_actor_indices: Apply func on a selected set of remote actors.
+            timeout_seconds: Ray.get() timeout. Default is None.
+                Note(jungong) : setting timeout_seconds to 0 effectively makes all the
+                remote calls fire-and-forget, while setting timeout_seconds to None
+                make them synchronous calls.
+
+        Returns:
+             The list of return values of all calls to `func(actor)`.
+        """
+        remote_actor_indices = remote_actor_indices or list(range(len(self.__actors)))
+        if healthy_only:
+            remote_actor_indices = [
+                i for i in remote_actor_indices if self.is_actor_healthy(i)
+            ]
+
+        remote_calls = self.__call_actors(
+            func=func,
+            remote_actor_indices=remote_actor_indices,
+        )
+
+        _, remote_results = self.__fetch_result_and_mark_state(
+            remote_actor_indices=remote_actor_indices,
+            remote_calls=remote_calls,
+            timeout_seconds=timeout_seconds,
+        )
+
+        if self._ignore_worker_failure:
```

Comment:
should ignore worker failure be a parameter of the class or a parameter passed to `foreach_actor`?